pkg/ocm/ocm.go

```diff
@@ -118,38 +118,40 @@ func (client Client) getClusterResource(clusterID string, resourceKey string) (s
 	return response.Body().Resources()[resourceKey], nil
 }
 
-// SendCHGMServiceLog allows to send a cluster has gone missing servicelog
-func (client Client) SendCHGMServiceLog(cluster *v1.Cluster) error {
+// SendCHGMServiceLog allows to send a cluster has gone missing servicelog.
+// On success it will return the sent service log entry.
+func (client Client) SendCHGMServiceLog(cluster *v1.Cluster) (*servicelog.LogEntry, error) {
 	json, err := getServiceLogTemplate(sl_clusterHasGoneMissing)
 	if err != nil {
-		return fmt.Errorf("failed to get CHGM-SL template: %w", err)
+		return nil, fmt.Errorf("failed to get CHGM-SL template: %w", err)
 	}
 	le, err := servicelog.UnmarshalLogEntry(json)
 	if err != nil {
-		return fmt.Errorf("failed to create SL message from json (%s): %w", json, err)
+		return nil, fmt.Errorf("failed to create SL message from json (%s): %w", json, err)
 	}
 	return client.sendServiceLog(le, cluster)
 }
 
-// sendServiceLog allows to send a generic servicelog to a cluster
-func (client Client) sendServiceLog(le *servicelog.LogEntry, cluster *v1.Cluster) error {
+// sendServiceLog allows to send a generic servicelog to a cluster.
+// On success it will return the sent service log entry for further processing.
+func (client Client) sendServiceLog(le *servicelog.LogEntry, cluster *v1.Cluster) (*servicelog.LogEntry, error) {
 	builder := servicelog.NewLogEntry()
```

Comment:
This needs to be updated to our new logic with the struct and its methods right ? instead of using the builder directly.